.github/workflows/release-build-native-linux.yml

```diff
@@ -0,0 +1,70 @@
+name: release-build-native-linux
+
+on:
+  release:
+    types: [created] 
+  push:
+    branches:
+      - master
+  pull_request:
+    branches:
+      - master
+
+jobs:
+# Trigger this job only on Release
+  build-on-release:
+    if: github.event_name=='release'&& github.event_name != 'pull_request'&& github.event_name != 'push'
+    name: Build on Release
+    runs-on: ubuntu-latest 
+
+    steps:
+    - uses: actions/checkout@v2
+    - name: Install graalvm
+      uses: DeLaGuardo/setup-graalvm@3
+      with:
+        graalvm-version: '20.0.0.java11'
+    - name: Install native-image
+      run: gu install native-image
+    - name: Set version
+      run: ./mvnw versions:set -DnewVersion="${{ github.event.release.tag_name }}" 
+    - name: Build native executable
+      run: ./mvnw package -Dnative 
+    - name: Upload native executable
+      id: upload-native-executable
+      uses: actions/upload-release-asset@v1
+      env:
+        GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+      with:
+        upload_url: ${{ github.event.release.upload_url }}
+        asset_path: ./target/starfish-${{ github.event.release.tag_name }}-runner
+        asset_name: starfish-linux
+        asset_content_type: application/octet-stream
+
+# Trigger this job only on Push or Pull-Request
+  build-on-push-pr:
+    if: github.event_name!='release'&& (github.event_name == 'pull_request'|| github.event_name == 'push')
+    name: Build on Push or PR
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - name: Install graalvm
+        uses: DeLaGuardo/setup-graalvm@3
+        with:
+          graalvm-version: '20.0.0.java11'
+      - name: Install native-image
+        run: gu install native-image
+      - name: Set version
+        run: ./mvnw versions:set -DnewVersion="1.0.0" 
```

Comment:
use pr number here ? or 999-SNAPSHOT

Reply:
Suppose I use a PR number and then after creating PR I again do some commit in my repo before the PR is merged,the workflow will have the same value.
Instead there is run_number that is unique number and increments every time the workflow is run: `${{ github.run_number }}`
- **GITHUB_RUN_NUMBER:** A unique number for each run of a particular workflow in a repository. This number begins at 1 for the workflow's first run, and increments with each new run. This number does not change if you re-run the workflow run.`${{ github.run_number }}`


Or 

There are other options as well:
- **GITHUB_SHA:** The commit SHA that triggered the workflow. For example, ffac537e6cbbf934b08745a378932722df287a53. Syntax: `${{ github.sha }}`

@maxandersen  I'm waiting for your thoughts :)

Reply:
Sha is the better number to use to differentiate

Reply:
I'll use SHA then!!